src/Stratis.Bitcoin.Features.RPC/IPAddressBlock.cs

```diff
@@ -79,15 +79,20 @@ public bool Contains(IPAddress address)
             if (this.MaskLength == (blockAddrBytes.Length * 8))
                 return this.BlockAddress.Equals(address);
 
+            // Determine the number of initial bytes in the block address that should match the ip address.
+            int maskBytesLength = this.MaskLength / 8;
             byte[] addressBytes = address.GetAddressBytes();
-            int bytesToKeep = this.MaskLength / 8;
-            if ((this.MaskLength % 8) != 0)
-                addressBytes[bytesToKeep++] &= (byte)~(0xff >> (this.MaskLength % 8));
+            for (int i = 0; i < maskBytesLength; i++)
+                if (addressBytes[i] != blockAddrBytes[i])
+                    return false;
 
-            while (bytesToKeep < addressBytes.Length)
-                addressBytes[bytesToKeep++] = 0;
+            // Deal with fractional bytes.
```

Comment:
Please add more description and maybe also a source to this calculations

Reply:
Done.